Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp

```diff
@@ -189,17 +193,23 @@ namespace GradientSignal
         m_configuration.m_imageAsset = asset;
     }
 
+    void ImageGradientComponent::OnGradientTransformChanged(const GradientTransform& newTransform)
+    {
+        AZStd::unique_lock<decltype(m_imageMutex)> lock(m_imageMutex);
+        m_gradientTransform = newTransform;
+    }
+
     float ImageGradientComponent::GetValue(const GradientSampleParams& sampleParams) const
     {
+        AZStd::shared_lock<decltype(m_imageMutex)> imageLock(m_imageMutex);
+
         AZ::Vector3 uvw = sampleParams.m_position;
 
         bool wasPointRejected = false;
-        GradientTransformRequestBus::Event(
-            GetEntityId(), &GradientTransformRequestBus::Events::TransformPositionToUVWNormalized, sampleParams.m_position, uvw, wasPointRejected);
+        m_gradientTransform.TransformPositionToUVWNormalized(sampleParams.m_position, uvw, wasPointRejected);
```

Comment:
Same comment as FastNoiseGradientComponent.cpp

Reply:
This one's a little bit different than the others, since the imageMutex is also needed for locking the image asset for the gradient lookup. i can still reduce scope to not include the initial uvw and wasPointRejected declarations though.